R/aldex.R

```diff
@@ -93,74 +86,30 @@ aldex <- function(Y, X, data=NULL, nsample=2000,  scale=NULL,
     formula <- NULL
   }
 
-  ## perform streaming 
   out <- list()
-  nsample.remaining <- nsample
   iter <- 1
-  if (stream) {
-    nsample.local <- floor(streamsize*100000/(N*D*8))
-    if (nsample.local < 1) stop("streamsize too small")
-  } else {
-    nsample.local <- nsample
+  chunk.sizes <- aldex.getchunksizes(Y, X, nsample, streamsize)
+  for(chunk.size in chunk.sizes) {
+    chunk <- aldex.chunk(Y, X, chunk.size, scale, scale.args,
+                         return.pars)
+    chunk <- c(chunk, fflm(aperm(chunk$logW, c(2,1,3)), t(X), test))
```

Comment:
To avoid memory overhead, can you just avoid defining `chunk` entirely. Instead modify `out`. So there is no unecessary copying.